exercise/static/exercise/chapter.js

```diff
@@ -579,31 +579,100 @@
           });
         });
       } else {
-        var exercise = this;
-        if (exercise.quiz) {
-          exercise.quizSubmission("submit");
-        } else {
-          /* TODO: migrate to overlay */
-          chapter.openModal(chapter.messages.submit);
+        // Callback for submitting an exercise
+        var submitCallback = function(exercise, hash) {
+          if (exercise.quiz) {
+            exercise.quizSubmission("submit");
+          } else {
+            /* TODO: migrate to overlay */
+            chapter.openModal(chapter.messages.submit);
+          }
+          var url = $(form_element).attr("action");
+          var formData = new FormData(form_element);
+          formData.append("__submission_hash", hash);
+
+          exercise.submitAjax(url, formData, function(data) {
+            //$(form_element).find(":input").prop("disabled", false);
+            //exercise.hideLoader();
+            var input = $(data);
+            if (exercise.quiz) {
+              exercise.quizSubmission("success");
+              exercise.update(input);
+              exercise.renderMath();
+              exercise.dom_element.dispatchEvent(
+                new CustomEvent("aplus:submission-finished",
+                  {bubbles: true, detail: {type: exercise.exercise_type}}));
+            } else {
+              exercise.updateSubmission(input);
+            }
+          });
+        };
+
+        let isDuplicate = false;
+        // Retrieve all the hashes of the previous submissions from the HTML
+        const hashes = new Set();
+        const datahashElements = this.element.find("[data-hash]");
+        for (let elem of datahashElements) {
+          try {
+            hashes.add(elem.dataset["hash"]);
+          } catch(error) {
+          }
         }
-        var url = $(form_element).attr("action");
-        var formData = new FormData(form_element);
 
-        exercise.submitAjax(url, formData, function(data) {
-          //$(form_element).find(":input").prop("disabled", false);
-          //exercise.hideLoader();
-          var input = $(data);
-          if (exercise.quiz) {
-            exercise.quizSubmission("success");
-            exercise.update(input);
-            exercise.renderMath();
-            exercise.dom_element.dispatchEvent(
-              new CustomEvent("aplus:submission-finished",
-                {bubbles: true, detail: {type: exercise.exercise_type}}));
+        var isDuplicateCallback = function(exercise, hash) {
+          isDuplicate = hashes.has(hash);
+          if (isDuplicate) {
+            var continueBtn = document.getElementById("duplicate-submission-modal-button");
+            continueBtn.onclick = () => {
+              submitCallback(exercise, hash);
+            }
```

Comment:
`onclick` is the old outdated API. Let's use the jQuery API here to attach the click event handler:
https://api.jquery.com/click/